Set show to expression

ggetz opened the pull request:
I set the `show` property in `Cesium3DTileStyle` to the expression specified in the json if it is a string. If it's a boolean, I create an expression with that boolean. If another type is supplied, I set `show` to `undefined`.

I added a spec for `Cesium3DTileStyle` where I check for setting the show property to various expressions.

In `Cesium3DTileStyleEngine` I also check if the show expression evaluates to a boolean.

ggetz commented:
I can't check if the expression evaluates to a boolean before runtime, because if it contains variables, it cannot be evaluated without a feature.

pjcozzi commented:
Update the Sandcastle example.

pjcozzi commented on Source/Scene/Cesium3DTileStyle.js:
```diff
@@ -66,11 +68,9 @@ define([
 
         var show;
         if (typeof(showExpression) === 'boolean') {
-            show = new LiteralBooleanExpression(styleEngine, showExpression);
```
Remove this file and `BooleanExpression`.

pjcozzi commented:
> I can't check if the expression evaluates to a boolean before runtime, because if it contains variables, it cannot be evaluated without a feature.

When I wrote the schema, I just required it to be a string, I think that is fine for now.  See https://github.com/AnalyticalGraphicsInc/3d-tiles/blob/style/Styling/schema/style.schema.json

pjcozzi commented:
Let's also add some tests that apply the style to a tileset, e.g., like https://github.com/AnalyticalGraphicsInc/cesium/blob/3d-tiles-style/Specs/Scene/Cesium3DTileBatchTableResourcesSpec.js#L447-L478

Add them to https://github.com/AnalyticalGraphicsInc/cesium/blob/3d-tiles-style/Specs/Scene/Cesium3DTilesetSpec.js

pjcozzi commented on Specs/Scene/Cesium3DTileStyleSpec.js:
```diff
@@ -0,0 +1,72 @@
+        var tileset = new MockTileset(styleEngine);
+
+        var style = new Cesium3DTileStyle(tileset, {
+            'show' : 'true'
```
Would probably just use `show` throughout instead of `'show'`.

ggetz commented:
Show I update the expression UI in the sandcastle example so that works?

pjcozzi commented:
Yes, just have it regenerate the entire expression string and create a new style until we decide when we are going to expose the AST.

ggetz commented:
> Let's also add some tests that apply the style to a tileset
>Add them to https://github.com/AnalyticalGraphicsInc/cesium/blob/3d-tiles-style/Specs/Scene/Cesium3DTilesetSpec.js

Should these tests be added to `Cesium3DTilesetSpecEngineSpec.js`, since that's where the style is actually applied?

pjcozzi commented:
I think it's fine to add them to Cesium3DTilesetSpec.js since the style engine is just an implementation detail.  If we do more fine-grained tests or we end up with a lot of tests, we can move them to a new file as you suggest.  Right now, I think it will only need a few tests.

ggetz commented:
Ok, I made the changes,  updated the sandcastle changes, and added the tests. I was a little unsure if the last test is set up right, but it is definitely not failing.

pjcozzi commented:
The test is OK, I added a few comments to clean it up.

Also, add one more test that uses a complex expression - including a variable - to get properties from the feature.  You'll want to use a different tileset that has a batch table and also create/destroy the scene in `beforeAll`/`afterAll` so the tests have better performance.

Perhaps this tileset is good for the new test:
```
var withBatchTableUrl = './Data/Cesium3DTiles/Batched/BatchedWithBatchTable/';

    it('renders tileset with batch table', function() {
        return Cesium3DTilesTester.loadTileset(scene, withBatchTableUrl).then(function(tileset) {
            var content = tileset._root.content;

            // Each feature in the b3dm file has an id property from 0 to 99,
            // check that the 2nd resource has an id of 2
            expect(content.getFeature(2).getProperty('id')).toEqual(2);

            // Check that a property can be an array
            expect(content.getFeature(2).getProperty('rooms')).toEqual(['room2_a', 'room2_b', 'room2_c']);

            // Check that a property can be an object
            expect(content.getFeature(2).getProperty('info')).toEqual({name : 'building2', year : 2});

            Cesium3DTilesTester.expectRenderTileset(scene, tileset);
        });
    });
```

ggetz commented:
I made the updates, and also moved the tests to https://github.com/AnalyticalGraphicsInc/cesium/blob/3d-tiles-style/Specs/Scene/Cesium3DTilesetSpec.js because I had accidentally put them in `Cesium3DTilesetStyleSpec.js`

pjcozzi commented:
:+1: Perfect!
